compiler/src/dotty/tools/dotc/ast/untpd.scala

```diff
@@ -76,8 +76,10 @@ object untpd extends Trees.Instance[Untyped] with UntypedTreeInfo {
   }
 
   /** A function type or closure with `implicit`, `erased`, or `given` modifiers */
-  class FunctionWithMods(args: List[Tree], body: Tree, val mods: Modifiers)(implicit @constructorOnly src: SourceFile)
-    extends Function(args, body)
+  class FunctionWithMods(args: List[Tree], body: Tree, val mods: Modifiers, val isErased: List[Boolean])(implicit @constructorOnly src: SourceFile)
```

Comment:
You should add a method `hasErasedParams`
```scala
def hasErasedParams: Boolean = isErased.contains(true)
```

and then replace all the `isErased.contains(true)` with `hasErasedParams`.

Should do something similar for `isErased.count(_ == false)`

Reply:
I think it's hard to do, since `isErased` is a list that comes from various places...